repos/system_upgrade/common/actors/checkmountoptions/libraries/checkmountoptions.py

```diff
@@ -0,0 +1,77 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def inhibit_upgrade_due_var_with_noexec(mountpoint, found_in_fstab=False):
+    summary = (
+        'Leapp detected that the {0} mountpoint is mounted with the "noexec" option, '
+        'which prevents binaries necessary for the upgrade from being executed. '
+        'The upgrade process cannot continue with {0} mounted using the "noexec" option.'
+    )
+
+    if found_in_fstab:
+        hint = (
+            'Temporarily remove the "noexec" option from {0} entry in /etc/fstab until the system is upgraded, '
+            'and remount the partition without the "noexec" option.'
+        )
+        related_resource = [reporting.RelatedResource('file', '/etc/fstab')]
+    else:
+        hint = 'Remount {0} without the noexec option.'
+        related_resource = []
+
+    reporting.create_report([
+        reporting.Title(
+            'The {0} mountpoint is mounted with the "noexec" option.'.format(mountpoint)
```

Comment:
@MichalHe reading the solution, I prefer to make it general s we do not have different "reports" for /var & /var/lib. Let's keep the title really static and make just dynamic other data (summary, remediation, ...). So the title could be e.g.
```suggestion
            'The partitions mounted with the "noexec" option preventing the correct in-place upgrade..'.format(mountpoint)
```

the function could accept the list of mountpoints instead of one.